omnibus/files/server-ctl-cookbooks/infra-server/templates/default/oc_erchef.config.erb

```diff
@@ -216,6 +216,14 @@
         {s3_url, "<%= @helper.bookshelf_s3_url %>"},
         <% end %>
         {s3_external_url, <%= @helper.erl_atom_or_string(node['private_chef']['bookshelf']['external_url']) %>},
+        %% bookshelf should automatically use path-style URLs.  s3 should automatically use vhost-style URLs unless s3_url_type is set otherwise.
+        <% if @helper.erl_atom_or_string(node['private_chef']['bookshelf']['external_url']).to_s == 'host_header' -%>
+        {s3_url_type, path},
+        <% elsif node['private_chef']['opscode-erchef']['s3_url_type'] -%>
+        {s3_url_type, <%= node['private_chef']['opscode-erchef']['s3_url_type'] %>},
```

Comment:
What would happen if the user set the s3_url_type to an invalid atom?

Reply:
If external_url is set to host_header, that will override, and the result will be path.  But if external_url is NOT set to host_header, it will render the invalid atom.  HOWEVEVER, that is caught in my erlang code, which forces it to either path or vhost.